doc/src/build/package-versioning.md

```diff
@@ -0,0 +1,30 @@
+---
+title: Package Versioning
+---
+
+Sui smart contracts are organized into [upgradeable](./package-upgrades.md) packages and, as a result, multiple versions of any given package can exist on-chain. Before a package can be used on the chain, it is [published](move/debug-publish.md#publishing-a-package) at its first (a.k.a, original) version. A new version of a package is created whenever the package is [upgraded](./package-upgrades.md), and an upgrade always happens with respect to the version of the package immediately preceding the upgraded one in the versions history. In other words, an upgrade to the Nth version of a package is always based on the N-1th version of the package (this is enforced by the system) and one can never create the Nth version of the package based on a version that's smaller than N-1th version. For example a package can be upgraded from version 1 to 2 but once this happens only an upgrade from version 2 to 3 is possible and an upgrade from version 1 to 3 is forbidden.
+
+Note that there is also a notion of versioning present in the [manifest](./move/manifest.md) files. You can observe it both in the [package section](./move/manifest.md#package-section) and in the [dependencies section](./move/manifest.md#dependencies-section). For example:
+```move
+[package]
+name = "some_pkg"
+version = "1.0.0"
+
+[dependencies]
+another_pkg = { git = "https://github.com/another_pkg/another_pkg.git" , version = "2.0.0"}
+```
+These fields in the manifest file, however, at this point are only used for user-level documentation purposes and are ignored by the build or publish/upgrade commands. Another way of looking at it is that if a developer publishes a package with a certain package version in the manifest file and then modifies and re-publishes the same package with a different version (using publish command rather than upgrade command), these two packages will be considered different packages, rather than on-chain versions of the same package, and recognized as such by the tooling - for example, none of these packages could be used as a [dependency override](./dependency-overrides.md) to stand in for the other one.
+
+## Version management
+
+In order to control the evolution of a published package, a developer can implement a [custom upgrade policy](./custom-upgrade-policy.md) but Sui also comes with a set of built-in upgrade policies. For an overview of how an upgrade policy should be applied, see [here](./custom-upgrade-policy.md#upgrade-overview).
```

Comment:
This statement makes it seem like "immutable", "dependency-only", "additive", "compatible" exist in the same class of things that people write custom upgrade policies for, which is not true, they are orthogonal.  This is my fault for overloading the world "policy", but maybe we can walk that back in the docs.

I think the best way to do that is to skip over the topic of custom upgrade policies here and talk just about the policies for what can be upgraded, maybe under the heading of "Compatibility"?  If we take this section and put it on the same page as the custom upgrade policy docs under that heading, that may suffice?

Reply:
I tried to finesse it as best I could.